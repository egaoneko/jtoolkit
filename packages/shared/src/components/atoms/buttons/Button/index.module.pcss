.button {
  @apply relative inline-flex items-center justify-center py-2.5 px-5 overflow-hidden;
  @apply rounded-md hover:shadow-lg focus:outline-none;
  @apply disabled:opacity-30 disabled:cursor-not-allowed;

  &.small {
    @apply text-sm;
  }

  &.medium {
    @apply text-base;
  }

  &.large {
    @apply text-lg;
  }

  &.rounded {
    @apply rounded-full;
  }

  &:not(:first-child) {
    @apply ml-2;
  }

  &.group {
    @apply ml-0 rounded-none;

    &:first-child {
      &.row {
        @apply rounded-l-md;
      }
      &.column {
        @apply rounded-t-md;
      }
    }

    &:last-child {
      &.row {
        @apply rounded-r-md;
      }
      &.column {
        @apply rounded-b-md;
      }
    }
  }

  & span.ripple {
    @apply absolute w-full h-full;

    & .effect {
      @apply absolute rounded-[50%] bg-white animate-ripple;
    }
  }
}

.contained {
  @apply text-white;

  &.primary {
    @apply bg-bgPrimaryDark hover:bg-bgPrimaryDarker;
    @apply dark:bg-dark-bgPrimaryDark hover:dark:bg-dark-bgPrimaryDarker;
  }

  &.secondary {
    @apply bg-bgSecondaryDark hover:bg-bgSecondaryDarker;
    @apply dark:bg-dark-bgSecondaryDark hover:dark:bg-dark-bgSecondaryDarker;
  }

  &.success {
    @apply bg-bgSuccessDark hover:bg-bgSuccessDarker;
    @apply dark:bg-dark-bgSuccessDark hover:dark:bg-dark-bgSuccessDarker;
  }

  &.danger {
    @apply bg-bgDangerDark hover:bg-bgDangerDarker;
    @apply dark:bg-dark-bgDangerDark hover:dark:bg-dark-bgDangerDarker;
  }

  &.warning {
    @apply bg-bgWarningDark hover:bg-bgWarningDarker;
    @apply dark:bg-dark-bgWarningDark hover:dark:bg-dark-bgWarningDarker;
  }

  &.info {
    @apply bg-bgInfoDark hover:bg-bgInfoDarker;
    @apply dark:bg-dark-bgInfoDark hover:dark:bg-dark-bgInfoDarker;
  }

  &.relief {
    @apply border-b-4;

    &.primary {
      @apply border-bgPrimaryDarker hover:bg-bgPrimary;
      @apply dark:border-dark-bgPrimaryDarker hover:dark:bg-dark-bgPrimary;
    }

    &.secondary {
      @apply border-bgSecondaryDarker hover:bg-bgSecondary;
      @apply dark:border-dark-bgSecondaryDarker hover:bg-bgSecondary;
    }

    &.success {
      @apply border-bgSuccessDarker hover:bg-bgSuccess;
      @apply dark:border-dark-bgSuccessDarker hover:bg-bgSuccess;
    }

    &.danger {
      @apply border-bgDangerDarker hover:bg-bgDanger;
      @apply dark:border-dark-bgDangerDarker hover:bg-bgDanger;
    }

    &.warning {
      @apply border-bgWarningDarker hover:bg-bgWarning;
      @apply dark:border-dark-bgWarningDarker hover:bg-bgWarning;
    }

    &.info {
      @apply border-bgInfoDarker hover:bg-bgInfo;
      @apply dark:border-dark-bgInfoDarker hover:bg-bgInfo;
    }
  }

  &.gradient {
    @apply bg-gradient-to-r;

    &.primary {
      @apply from-bgPrimaryLight to-bgPrimaryDarker;
      @apply dark:from-bgPrimaryLight dark:to-bgPrimaryDarker;
    }

    &.secondary {
      @apply from-bgSecondaryLight to-bgSecondaryDarker;
      @apply dark:from-bgSecondaryLight dark:to-bgSecondaryDarker;
    }

    &.success {
      @apply from-bgSuccessLight to-bgSuccessDarker;
      @apply dark:from-bgSuccessLight dark:to-bgSuccessDarker;
    }

    &.danger {
      @apply from-bgDangerLight to-bgDangerDarker;
      @apply dark:from-bgDangerLight dark:to-bgDangerDarker;
    }

    &.warning {
      @apply from-bgWarningLight to-bgWarningDarker;
      @apply dark:from-bgWarningLight dark:to-bgWarningDarker;
    }

    &.info {
      @apply from-bgInfoLight to-bgInfoDarker;
      @apply dark:from-bgInfoLight dark:to-bgInfoDarker;
    }
  }
}

.outlined {
  @apply border;

  &.primary {
    @apply text-textPrimaryDarker border-bgPrimaryDarker hover:bg-bgPrimaryLighter/[.05];
    @apply dark:text-dark-textPrimaryDarker dark:border-dark-bgPrimaryDarker hover:dark:bg-dark-bgPrimaryLighter/[.05];
  }

  &.secondary {
    @apply text-textSecondaryDarker border-bgSecondaryDarker hover:bg-bgSecondaryLighter/[.05];
    @apply dark:text-dark-textSecondaryDarker dark:border-dark-bgSecondaryDarker hover:dark:bg-dark-bgSecondaryLighter/[.05];
  }

  &.success {
    @apply text-textSuccessDarker border-bgSuccessDarker hover:bg-bgSuccessLighter/[.05];
    @apply dark:text-dark-textSuccessDarker dark:border-dark-bgSuccessDarker hover:dark:bg-dark-bgSuccessLighter/[.05];
  }

  &.danger {
    @apply text-textDangerDarker border-bgDangerDarker hover:bg-bgDangerLighter/[.05];
    @apply dark:text-dark-textDangerDarker dark:border-dark-bgDangerDarker hover:dark:bg-dark-bgDangerLighter/[.05];
  }

  &.warning {
    @apply text-textWarningDarker border-bgWarningDarker hover:bg-bgWarningLighter/[.05];
    @apply dark:text-dark-textWarningDarker dark:border-dark-bgWarningDarker hover:dark:bg-dark-bgWarningLighter/[.05];
  }

  &.info {
    @apply text-textInfoDarker border-bgInfoDarker hover:bg-bgInfoLighter/[.05];
    @apply dark:text-dark-textInfoDarker dark:border-dark-bgInfoDarker hover:dark:bg-dark-bgInfoLighter/[.05];
  }
}

.text {
  &.primary {
    @apply text-textPrimaryDarker hover:bg-bgPrimaryLight/[.1];
    @apply dark:text-dark-textPrimaryDarker hover:dark:bg-dark-bgPrimaryLight/[.1];
  }

  &.secondary {
    @apply text-textSecondaryDarker hover:bg-bgSecondaryLight/[.1];
    @apply dark:text-dark-textSecondaryDarker hover:dark:bg-dark-bgSecondaryLight/[.1];
  }

  &.success {
    @apply text-textSuccessDarker hover:bg-bgSuccessLight/[.1];
    @apply dark:text-dark-textSuccessDarker hover:dark:bg-dark-bgSuccessLight/[.1];
  }

  &.danger {
    @apply text-textDangerDarker hover:bg-bgDangerLight/[.1];
    @apply dark:text-dark-textDangerDarker hover:dark:bg-dark-bgDangerLight/[.1];

  }

  &.warning {
    @apply text-textWarningDarker hover:bg-bgWarningLight/[.1];
    @apply dark:text-dark-textWarningDarker hover:dark:bg-dark-bgWarningLight/[.1];
  }

  &.info {
    @apply text-textInfoDarker hover:bg-bgInfoLight/[.1];
    @apply dark:text-dark-textInfoDarker hover:dark:bg-dark-bgInfoLight/[.1];
  }
}

