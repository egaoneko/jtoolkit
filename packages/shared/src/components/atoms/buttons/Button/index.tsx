import { ButtonHTMLAttributes, MouseEvent, FC, useCallback, useRef, ReactNode } from 'react';
import clsx from 'clsx';

import { Color } from '../../../../enums/color';
import { ButtonGroupDirection, ButtonVariant } from '../../../../enums/button';
import { Size } from '../../../../enums/size';
import Ripple from '../../effect/Ripple';

import styles from './index.module.pcss';

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  size?: Size;
  color?: Color;
  disabled?: boolean;
  rounded?: boolean;
  popped?: boolean;
  gradient?: boolean;
  relief?: boolean;
  ripple?: boolean;
  rippleCenter?: boolean;
  group?: boolean;
  groupDirection?: ButtonGroupDirection;
  startIcon?: ReactNode;
  endIcon?: ReactNode;
  className?: string;
}

const Button: FC<ButtonProps> = ({
  variant = ButtonVariant.CONTAINED,
  size = Size.MEDIUM,
  color = Color.PRIMARY,
  disabled = false,
  rounded = false,
  popped = false,
  gradient = false,
  relief = false,
  ripple = true,
  rippleCenter = false,
  group = false,
  groupDirection = ButtonGroupDirection.ROW,
  startIcon,
  endIcon,
  onClick,
  className,
  children,
  ...props
}) => {
  const buttonRef = useRef<HTMLButtonElement>(null);

  const handleOnClick = useCallback(
    (event: MouseEvent<HTMLButtonElement>) => {
      onClick?.(event);
    },
    [onClick],
  );

  return (
    <button
      ref={buttonRef}
      className={clsx(
        styles.button,
        styles[variant],
        styles[size],
        styles[color],
        {
          [styles.rounded]: rounded,
          [styles.gradient]: gradient,
          [styles.relief]: relief,
          [styles.group]: group,
          [styles[groupDirection]]: group,
          ['transform hover:scale-110']: !group && (popped || gradient),
        },
        className,
      )}
      disabled={disabled}
      onClick={handleOnClick}
      {...props}
    >
      {startIcon && <span>{startIcon}</span>}
      {children}
      {endIcon && <span>{endIcon}</span>}
      {ripple && <Ripple parentRef={buttonRef} center={rippleCenter} />}
    </button>
  );
};

export default Button;
