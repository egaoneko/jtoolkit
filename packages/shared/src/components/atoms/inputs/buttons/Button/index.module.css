.button {
  @apply relative inline-flex items-center justify-center;
  @apply box-border overflow-hidden select-none;
  @apply rounded-md hover:shadow-lg focus:outline-none;
  @apply disabled:opacity-30 disabled:cursor-not-allowed;

  &.doubleExtraSmall {
    @apply text-xs py-1.5 px-3 gap-1;
  }

  &.extraSmall {
    @apply text-sm py-2 px-4 gap-1.5;
  }

  &.small {
    @apply text-lg py-2.5 px-5 gap-2;
  }

  &.medium {
    @apply text-2xl py-3 px-6 gap-2.5;
  }

  &.large {
    @apply text-4xl py-3.5 px-7 gap-3;
  }

  &.extraLarge {
    @apply text-6xl py-4 px-8 gap-3.5;
  }

  &.doubleExtraLarge {
    @apply text-8xl py-5 px-10 gap-4;
  }

  &.rounded {
    @apply rounded-full;
  }

  &:not(:first-child) {
    @apply ml-2;
  }

  &.group {
    @apply ml-0 rounded-none;

    &:first-child {
      &.row {
        @apply rounded-l-md;
      }

      &.column {
        @apply rounded-t-md;
      }
    }

    &:last-child {
      &.row {
        @apply rounded-r-md;
      }

      &.column {
        @apply rounded-b-md;
      }
    }
  }

  & span.ripple {
    @apply absolute w-full h-full;

    & .effect {
      @apply absolute rounded-[50%] bg-white animate-ripple;
    }
  }
}

.contained {
  @apply text-white;

  &.primary {
    @apply bg-bgPrimaryDark hover:bg-bgPrimaryDarker;
    @apply dark:bg-dark-bgPrimaryDark hover:dark:bg-dark-bgPrimaryDarker;
  }

  &.secondary {
    @apply bg-bgSecondaryDark hover:bg-bgSecondaryDarker;
    @apply dark:bg-dark-bgSecondaryDark hover:dark:bg-dark-bgSecondaryDarker;
  }

  &.success {
    @apply bg-bgSuccessDark hover:bg-bgSuccessDarker;
    @apply dark:bg-dark-bgSuccessDark hover:dark:bg-dark-bgSuccessDarker;
  }

  &.danger {
    @apply bg-bgDangerDark hover:bg-bgDangerDarker;
    @apply dark:bg-dark-bgDangerDark hover:dark:bg-dark-bgDangerDarker;
  }

  &.warning {
    @apply bg-bgWarningDark hover:bg-bgWarningDarker;
    @apply dark:bg-dark-bgWarningDark hover:dark:bg-dark-bgWarningDarker;
  }

  &.info {
    @apply bg-bgInfoDark hover:bg-bgInfoDarker;
    @apply dark:bg-dark-bgInfoDark hover:dark:bg-dark-bgInfoDarker;
  }

  &.relief {
    @apply border-b-4;

    &.primary {
      @apply border-bgPrimaryDarker hover:bg-bgPrimary;
      @apply dark:border-dark-bgPrimaryDarker hover:dark:bg-dark-bgPrimary;
    }

    &.secondary {
      @apply border-bgSecondaryDarker hover:bg-bgSecondary;
      @apply dark:border-dark-bgSecondaryDarker hover:bg-bgSecondary;
    }

    &.success {
      @apply border-bgSuccessDarker hover:bg-bgSuccess;
      @apply dark:border-dark-bgSuccessDarker hover:bg-bgSuccess;
    }

    &.danger {
      @apply border-bgDangerDarker hover:bg-bgDanger;
      @apply dark:border-dark-bgDangerDarker hover:bg-bgDanger;
    }

    &.warning {
      @apply border-bgWarningDarker hover:bg-bgWarning;
      @apply dark:border-dark-bgWarningDarker hover:bg-bgWarning;
    }

    &.info {
      @apply border-bgInfoDarker hover:bg-bgInfo;
      @apply dark:border-dark-bgInfoDarker hover:bg-bgInfo;
    }
  }

  &.gradient {
    @apply bg-gradient-to-r;

    &.primary {
      @apply from-bgPrimaryLight to-bgPrimaryDarker;
      @apply dark:from-bgPrimaryLight dark:to-bgPrimaryDarker;
    }

    &.secondary {
      @apply from-bgSecondaryLight to-bgSecondaryDarker;
      @apply dark:from-bgSecondaryLight dark:to-bgSecondaryDarker;
    }

    &.success {
      @apply from-bgSuccessLight to-bgSuccessDarker;
      @apply dark:from-bgSuccessLight dark:to-bgSuccessDarker;
    }

    &.danger {
      @apply from-bgDangerLight to-bgDangerDarker;
      @apply dark:from-bgDangerLight dark:to-bgDangerDarker;
    }

    &.warning {
      @apply from-bgWarningLight to-bgWarningDarker;
      @apply dark:from-bgWarningLight dark:to-bgWarningDarker;
    }

    &.info {
      @apply from-bgInfoLight to-bgInfoDarker;
      @apply dark:from-bgInfoLight dark:to-bgInfoDarker;
    }
  }
}

.outlined {
  @apply border;

  &.primary {
    @apply text-textPrimaryDarker border-bgPrimaryDarker hover:bg-bgPrimaryLighter/[0.05];
    @apply dark:text-dark-textPrimaryDarker dark:border-dark-bgPrimaryDarker hover:dark:bg-dark-bgPrimaryLighter/[0.05];
  }

  &.secondary {
    @apply text-textSecondaryDarker border-bgSecondaryDarker hover:bg-bgSecondaryLighter/[0.05];
    @apply dark:text-dark-textSecondaryDarker dark:border-dark-bgSecondaryDarker hover:dark:bg-dark-bgSecondaryLighter/[0.05];
  }

  &.success {
    @apply text-textSuccessDarker border-bgSuccessDarker hover:bg-bgSuccessLighter/[0.05];
    @apply dark:text-dark-textSuccessDarker dark:border-dark-bgSuccessDarker hover:dark:bg-dark-bgSuccessLighter/[0.05];
  }

  &.danger {
    @apply text-textDangerDarker border-bgDangerDarker hover:bg-bgDangerLighter/[0.05];
    @apply dark:text-dark-textDangerDarker dark:border-dark-bgDangerDarker hover:dark:bg-dark-bgDangerLighter/[0.05];
  }

  &.warning {
    @apply text-textWarningDarker border-bgWarningDarker hover:bg-bgWarningLighter/[0.05];
    @apply dark:text-dark-textWarningDarker dark:border-dark-bgWarningDarker hover:dark:bg-dark-bgWarningLighter/[0.05];
  }

  &.info {
    @apply text-textInfoDarker border-bgInfoDarker hover:bg-bgInfoLighter/[0.05];
    @apply dark:text-dark-textInfoDarker dark:border-dark-bgInfoDarker hover:dark:bg-dark-bgInfoLighter/[0.05];
  }
}

.text {
  &.primary {
    @apply text-textPrimaryDarker hover:bg-bgPrimaryLight/[0.1];
    @apply dark:text-dark-textPrimaryDarker hover:dark:bg-dark-bgPrimaryLight/[0.1];
  }

  &.secondary {
    @apply text-textSecondaryDarker hover:bg-bgSecondaryLight/[0.1];
    @apply dark:text-dark-textSecondaryDarker hover:dark:bg-dark-bgSecondaryLight/[0.1];
  }

  &.success {
    @apply text-textSuccessDarker hover:bg-bgSuccessLight/[0.1];
    @apply dark:text-dark-textSuccessDarker hover:dark:bg-dark-bgSuccessLight/[0.1];
  }

  &.danger {
    @apply text-textDangerDarker hover:bg-bgDangerLight/[0.1];
    @apply dark:text-dark-textDangerDarker hover:dark:bg-dark-bgDangerLight/[0.1];
  }

  &.warning {
    @apply text-textWarningDarker hover:bg-bgWarningLight/[0.1];
    @apply dark:text-dark-textWarningDarker hover:dark:bg-dark-bgWarningLight/[0.1];
  }

  &.info {
    @apply text-textInfoDarker hover:bg-bgInfoLight/[0.1];
    @apply dark:text-dark-textInfoDarker hover:dark:bg-dark-bgInfoLight/[0.1];
  }
}
