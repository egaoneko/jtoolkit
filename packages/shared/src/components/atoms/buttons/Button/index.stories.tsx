import { Story } from '@storybook/react';

import { Color } from '../../../../enums/color';
import { ButtonVariant } from '../../../../enums/button';
import { Size } from '../../../../enums/size';

import Button, { ButtonProps } from './index';

export default {
  title: 'atoms/buttons/Button',
  component: Button,
};

const StandardTemplate: Story<ButtonProps> = args => (
  <div>
    <Button {...args} variant={ButtonVariant.CONTAINED}>
      Contained
    </Button>
    <Button {...args} variant={ButtonVariant.OUTLINED}>
      Outlined
    </Button>
    <Button {...args} variant={ButtonVariant.TEXT}>
      Text
    </Button>
  </div>
);

export const Standard = StandardTemplate.bind({});
Standard.args = {
  size: Size.DEFAULT,
  color: Color.PRIMARY,
  disabled: false,
  rounded: false,
  popped: false,
  gradient: false,
  relief: false,
  ripple: true,
};
