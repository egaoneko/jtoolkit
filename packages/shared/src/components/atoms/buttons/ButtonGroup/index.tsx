import { cloneElement, FC } from 'react';
import clsx from 'clsx';

import { ButtonGroupDirection } from '../../../../enums/button';
import toChildrenArray from '../../../../utils/element';
import { ButtonProps } from '../Button';

import styles from './index.module.pcss';

export interface ButtonGroupProps {
  direction?: ButtonGroupDirection;
}

const ButtonGroup: FC<ButtonGroupProps> = ({ direction = ButtonGroupDirection.ROW, children }) => {
  const buttons = toChildrenArray<ButtonProps>(children);
  return <div className={clsx(styles.group, styles[direction])}>{renderButtons(buttons, direction)}</div>;
};

export default ButtonGroup;

function renderButtons(children: React.ReactElement<ButtonProps>[], direction: ButtonGroupDirection): React.ReactNode {
  return children.map(child => cloneElement(child, { ...child.props, group: true, groupDirection: direction }));
}
