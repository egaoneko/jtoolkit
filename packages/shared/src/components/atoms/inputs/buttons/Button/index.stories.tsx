import { Story } from '@storybook/react';

import { Color } from '../../../../../enums/color';
import { ButtonVariant } from '../../../../../enums/button';
import { Size } from '../../../../../enums/size';
import SvgIcon from '../../../displays/icons/SvgIcon';

import Button, { ButtonProps } from './index';

export default {
  title: 'atoms/inputs/buttons/Button',
  component: Button,
};

const StandardTemplate: Story<ButtonProps> = args => (
  <div className="flex gap-2">
    <Button {...args} variant={ButtonVariant.CONTAINED}>
      Contained
    </Button>
    <Button {...args} variant={ButtonVariant.OUTLINED}>
      Outlined
    </Button>
    <Button {...args} variant={ButtonVariant.TEXT}>
      Text
    </Button>
  </div>
);

export const Standard = StandardTemplate.bind({});
Standard.args = {
  size: Size.MEDIUM,
  color: Color.PRIMARY,
  disabled: false,
  rounded: false,
  popped: false,
  gradient: false,
  relief: false,
  ripple: true,
  rippleCenter: false,
};

const icon = (
  <SvgIcon>
    <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
  </SvgIcon>
);
const IconTemplate: Story<ButtonProps> = args => (
  <div className="flex gap-2">
    <Button {...args} startIcon={icon}>
      Start
    </Button>
    <Button {...args} endIcon={icon}>
      End
    </Button>
  </div>
);

export const Icon = IconTemplate.bind({});
Icon.args = {
  size: Size.MEDIUM,
  color: Color.PRIMARY,
  variant: ButtonVariant.CONTAINED,
};
