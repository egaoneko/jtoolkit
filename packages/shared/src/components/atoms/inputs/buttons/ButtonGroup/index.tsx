import { cloneElement, FC } from 'react';
import clsx from 'clsx';

import { ButtonGroupDirection } from 'src/enums/button';
import toChildrenArray from 'src/utils/element';
import { ButtonProps } from 'src/components/atoms/inputs/buttons/Button';

import styles from './index.module.css';

export interface ButtonGroupProps {
  direction?: ButtonGroupDirection;
}

const ButtonGroup: FC<ButtonGroupProps> = ({ direction = ButtonGroupDirection.ROW, children }) => {
  const buttons = toChildrenArray<ButtonProps>(children);
  return <div className={clsx(styles.group, styles[direction])}>{renderButtons(buttons, direction)}</div>;
};

export default ButtonGroup;

function renderButtons(children: React.ReactElement<ButtonProps>[], direction: ButtonGroupDirection): React.ReactNode {
  return children.map((child, index) =>
    cloneElement(child, { key: `__buttons-${index}`, group: true, groupDirection: direction, ...child.props }),
  );
}
