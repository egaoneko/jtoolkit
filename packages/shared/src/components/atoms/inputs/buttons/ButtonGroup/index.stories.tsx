import { Story } from '@storybook/react';

import { ButtonGroupDirection } from '../../../../../enums/button';
import Button from '../Button';

import ButtonGroup, { ButtonGroupProps } from './index';

export default {
  title: 'atoms/inputs/buttons/ButtonGroup',
  component: ButtonGroup,
};

const StandardTemplate: Story<ButtonGroupProps> = args => (
  <ButtonGroup {...args}>
    <Button>One</Button>
    <Button>Two</Button>
    <Button>Three</Button>
  </ButtonGroup>
);

export const Standard = StandardTemplate.bind({});
Standard.args = {
  direction: ButtonGroupDirection.ROW,
};
