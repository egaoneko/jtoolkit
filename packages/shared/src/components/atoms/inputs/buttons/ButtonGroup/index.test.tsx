import renderer, { ReactTestInstance } from 'react-test-renderer';

import Button from 'src/components/atoms/inputs/buttons/Button/index';
import ButtonGroup from 'src/components/atoms/inputs/buttons/ButtonGroup/index';
import { ButtonGroupDirection } from 'src/enums';

describe('ButtonGroup Snapshots', () => {
  test('renders ButtonGroup with default options', async () => {
    const tree = renderer.create(
      <ButtonGroup>
        <Button>Test1</Button>
        <Button>Test2</Button>
      </ButtonGroup>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[0] as ReactTestInstance).children[0] as ReactTestInstance)
        .type,
    ).toBe('button');
    expect(
      (((tree.root.children[0] as ReactTestInstance).children[1] as ReactTestInstance).children[0] as ReactTestInstance)
        .type,
    ).toBe('button');
    expect(tree.toTree()?.props.className).toContain('row');
  });

  test('renders ButtonGroup with column', async () => {
    const tree = renderer.create(
      <ButtonGroup direction={ButtonGroupDirection.COLUMN}>
        <Button>Test1</Button>
        <Button>Test2</Button>
      </ButtonGroup>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(tree.toTree()?.props.className).toContain('column');
  });
});
