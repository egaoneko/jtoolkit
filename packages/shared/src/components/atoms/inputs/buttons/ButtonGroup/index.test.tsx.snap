// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Snapshots renders ButtonGroup with column 1`] = `
<div
  className="group column"
>
  <button
    className="button contained medium primary group column"
    disabled={false}
  >
    Test1
    <span
      className="ripple"
    />
  </button>
  <button
    className="button contained medium primary group column"
    disabled={false}
  >
    Test2
    <span
      className="ripple"
    />
  </button>
</div>
`;

exports[`ButtonGroup Snapshots renders ButtonGroup with default options 1`] = `
<div
  className="group row"
>
  <button
    className="button contained medium primary group row"
    disabled={false}
  >
    Test1
    <span
      className="ripple"
    />
  </button>
  <button
    className="button contained medium primary group row"
    disabled={false}
  >
    Test2
    <span
      className="ripple"
    />
  </button>
</div>
`;
